File: src/main/java/ssg/search/base/Collection.java
package ssg.search.base;

import ssg.search.parameter.Parameter;
import ssg.search.result.Result;
import QueryAPI510.Search;

public interface Collection {
	public String getCollectionName(Parameter parameter);
	public String getCollectionAliasName(Parameter parameter);
	public String[] getDocumentField(Parameter parameter);
	public String[] getSearchField(Parameter parameter);
	public Result getResult(Search search, String name, Parameter parameter, Result result);
}

File: src/main/kotlin/ssg/front/search/api/collection/Collection.kt
package ssg.front.search.api.collection

import QueryAPI510.Search
import ssg.front.search.api.dto.Parameter
import ssg.front.search.api.dto.result.Result

interface Collection {
    fun getName(parameter: Parameter): String?
    fun getResult(search: Search, name: String, parameter: Parameter, result: Result): Result
}